NSURL.m

```diff
@@ -351,7 +352,25 @@ static CFTypeRef CFURLCreatePropertyForKey(CFURLRef url, CFStringRef key, CFErro
     else if (CFEqual(key, kCFURLTypeIdentifierKey))
     {
         // Key for the resource’s uniform type identifier (UTI), returned as a CFString object.
+        void* LaunchServicesHandle = dlopen("/System/Library/Frameworks/CoreServices.framework/Frameworks/LaunchServices.framework/LaunchServices", RTLD_LAZY | RTLD_LOCAL);
+        if(LaunchServicesHandle == NULL){
+            posixError(error);
+            return nil;
+        }
+        CFStringRef (*UTTypeCreatePreferredIdentifierForTag)(CFStringRef, CFStringRef, _Nullable CFStringRef) = dlsym(LaunchServicesHandle, "UTTypeCreatePreferredIdentifierForTag");
+        CFStringRef (*tagType) = dlsym(LaunchServicesHandle, "kUTTagClassFilenameExtension");
+        if(UTTypeCreatePreferredIdentifierForTag == NULL || tagType == NULL){
+            dlclose(LaunchServicesHandle);
+            posixError(error);
+        }
+        
+        CFStringRef extension = CFURLCopyPathExtension(url);
+        value = UTTypeCreatePreferredIdentifierForTag((*tagType), extension, nil); 
 
+        int closed = dlclose(LaunchServicesHandle);
+        if(close != 0){
+            posixError(error);
+        }
```

Comment:
And you're leaking `extension`, since you create it with a "copy" function (remember the [create rule](https://developer.apple.com/library/archive/documentation/CoreFoundation/Conceptual/CFMemoryMgmt/Concepts/Ownership.html#//apple_ref/doc/uid/20001148-103029)!)